src/app/lib/analyticsUtils/index.test.js

```diff
@@ -570,3 +571,71 @@ describe('getThingAttributes', () => {
     expect(thingAttributes).toEqual(null);
   });
 });
+
+describe('getCampaignType', () => {
+  const windowLocation = window.location;
+
+  afterEach(() => {
+    resetWindowValue('location', windowLocation);
+  });
+
+  it('should return campaign type of email', () => {
+    setWindowValue('location', {
+      href: 'https://www.bbc.com/mundo?at_medium=email',
+    });
+
+    const campaignType = getCampaignType();
+
+    expect(campaignType).toEqual('email');
+  });
+
+  it('should return campaign type of affiliate', () => {
+    setWindowValue('location', {
+      href: 'https://www.bbc.com/mundo?at_medium=affiliate',
+    });
+
+    const campaignType = getCampaignType();
+
+    expect(campaignType).toEqual('affiliate');
+  });
+
+  it('should return campaign type of custom123', () => {
+    setWindowValue('location', {
+      href: 'https://www.bbc.com/mundo?at_medium=custom123',
+    });
+
+    const campaignType = getCampaignType();
+
+    expect(campaignType).toEqual('custom123');
+  });
+
+  it('should return null for invalid campaign type', () => {
+    setWindowValue('location', {
+      href: 'https://www.bbc.com/mundo?at_medium=foo',
+    });
+
+    const campaignType = getCampaignType();
+
+    expect(campaignType).toEqual(null);
+  });
+
+  it('should return campaign type of XTOR', () => {
+    setWindowValue('location', {
+      href: 'https://www.bbc.com/mundo?xtor=123',
+    });
+
+    const campaignType = getCampaignType();
+
+    expect(campaignType).toEqual('XTOR');
+  });
+
+  it('should return campaign type of XTOR', () => {
+    setWindowValue('location', {
+      href: 'https://www.bbc.com/mundo#xtor',
+    });
+
+    const campaignType = getCampaignType();
+
+    expect(campaignType).toEqual('XTOR');
+  });
```

Comment:
Could these be done as `it.each` style tests to reduce the duplication in the tests: https://jestjs.io/docs/en/api#testeachtablename-fn-timeout

Reply:
Updated, not used this syntax here before but I like it 👍